File: src/main/java/model/OrderItem.java
package model;

import jakarta.persistence.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.Nationalized;

import java.math.BigDecimal;
import java.util.UUID;

@Entity
@Table(name = "OrderItems")
public class OrderItem {
    @Id
    @ColumnDefault("newid()")
    @Column(name = "OrderItemID", nullable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "OrderID", nullable = false)
    private Order orderID;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "ProductID", nullable = false)
    private Product productID;

    @Nationalized
    @Column(name = "ProductName", nullable = false)
    private String productName;

    @Nationalized
    @Column(name = "SKU", nullable = false, length = 100)
    private String sku;

    @Column(name = "Quantity", nullable = false)
    private Integer quantity;

    @Column(name = "UnitPrice", nullable = false, precision = 18, scale = 2)
    private BigDecimal unitPrice;

    @Column(name = "TotalPrice", nullable = false, precision = 18, scale = 2)
    private BigDecimal totalPrice;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public Order getOrderID() {
        return orderID;
    }

    public void setOrderID(Order orderID) {
        this.orderID = orderID;
    }

    public Product getProductID() {
        return productID;
    }

    public void setProductID(Product productID) {
        this.productID = productID;
    }

    public String getProductName() {
        return productName;
    }

    public void setProductName(String productName) {
        this.productName = productName;
    }

    public String getSku() {
        return sku;
    }

    public void setSku(String sku) {
        this.sku = sku;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }

    public BigDecimal getUnitPrice() {
        return unitPrice;
    }

    public void setUnitPrice(BigDecimal unitPrice) {
        this.unitPrice = unitPrice;
    }

    public BigDecimal getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(BigDecimal totalPrice) {
        this.totalPrice = totalPrice;
    }

}

File: src/main/java/model/OrderItem.kt
package model

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import org.hibernate.annotations.Nationalized
import java.math.BigDecimal
import java.util.*

@Entity
@Table(name = "OrderItems")
open class OrderItem {
    @Id
    @ColumnDefault("newid()")
    @Column(name = "OrderItemID", nullable = false)
    open var id: UUID? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "OrderID", nullable = false)
    open var orderID: Order? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "ProductID", nullable = false)
    open var productID: Product? = null

    @Nationalized
    @Column(name = "ProductName", nullable = false)
    open var productName: String? = null

    @Nationalized
    @Column(name = "SKU", nullable = false, length = 100)
    open var sku: String? = null

    @Column(name = "Quantity", nullable = false)
    open var quantity: Int? = null

    @Column(name = "UnitPrice", nullable = false, precision = 18, scale = 2)
    open var unitPrice: BigDecimal? = null

    @Column(name = "TotalPrice", nullable = false, precision = 18, scale = 2)
    open var totalPrice: BigDecimal? = null
}